src/core/layertree/qgslayertreemodellegendnode.cpp

```diff
@@ -739,12 +743,15 @@ QSizeF QgsSymbolLegendNode::drawSymbol( const QgsLegendSettings &settings, ItemC
     // QGIS 4.0 -- ctx->context will be mandatory
     const bool useAdvancedEffects = ctx->context ? ctx->context->flags() & Qgis::RenderContextFlag::UseAdvancedEffects : settings.useAdvancedEffects();
     Q_NOWARN_DEPRECATED_POP
-    if ( opacity != 255 && useAdvancedEffects )
+    if ( !restrictedSizeSymbolOK || ( opacity != 255 && useAdvancedEffects ) )
```

Comment:
Instead of using the rasterisation code path, could  a clipping path be set on the painter instead? That'd avoid the potentially unwanted forced rasterisation

Reply:
Any update on this ? I'm ok to use clipping path if you do think it is worthy regarding my previous comment